pkg/controllers/vrpq/query_reconciler.go

```diff
@@ -108,8 +108,15 @@ func (q *QueryReconciler) collectInfoFromVdb(ctx context.Context) (ctrl.Result,
 // Temporarily, runListRestorePoints will not call the ListRestorePoints API
 // since the dispatcher is not set up yet
 func (q *QueryReconciler) runListRestorePoints(ctx context.Context, _ *ctrl.Request) error {
+	// update query message prior to calling vclusterops API
+	err := vrpqstatus.UpdateMessageStatus(ctx, q.VRec.Client, q.VRec.Log, q.Vrpq,
+		"Query is still in progress")
```

Comment:
I agree with you that we should add a defer statement to always reset the status. However, I'm uncertain about the message content. Should we name it 'reset query'? My concern is that if the defer call sets the state for both success and failure, when we execute 'kubectl get vrpq', it will display:
```                                                                                                                                           
"NAME   AGE     MESSAGE
  vc     5m21s   reset query
``
We won't be able to distinguish whether it was successful or failed

Reply:
We could always set the `Querying` condition to `False` (if it is not already) when we exit?
As for the `Message/State` I think we could set its value depending on its current value? If its current value is "Query is still in progress" it means the query hasn't been successfully completed yet, and we could say something like "Query failed". If its current value is "Query has completed successfully" we have no need to update it.

Reply:
Isn't `defer` more suitable for cleanup operations rather than modifying values? It can make the code more complex. Although you can have `defer` call a function pass a parameter(maybe an error) and do what you have to do, but we would need to change the returned value in the defer call. Any reason why we cannot set success/failure whithin the function body based on the API call result?

Reply:
@jizhuoyu I wonder what current value are you mentioning now? `Query is still in progress` and `Query has completed successfully` are messages we update in message field. As @spilchen mentioned, there are 3 messages we can update  `Querying`, `Queried successful` and `Query failed`. 

@roypaulin @spilchen since we are currently discussing whether to use a deferred function or not, I will wait until we reach an agreement

Reply:
@chinhtranvan I mean you could always first get the current value of message/state field via the k8s client, right? And you could then update the field accordingly.

Reply:
@chinhtranvan This only makes sense if we want to use a single `defer` statement to handle all error paths within the function body, though.

Reply:
I see the concern with using the defer. I think I would feel better if there were less error code paths in `runListRestorePoints`. Could we combine the updates to the status condition and the state in the same call? Then it really cuts down on the error flows. We kind of already set the Querying status condition in lock step with the state. So, it would make sense to update them at the same time.